File: searchsort/insertion.go
package main

import "fmt"

func main() {
	sample := []int{2, 5, 3, 4, 7, 1}
	insertion(sample)

}

func insertion(sample []int) {
	n := len(sample)
	for i := 1; i < n; i++ {
		a := sample[i]
		pos := i
		for pos > 0 && sample[pos-1] > a {
			sample[pos] = sample[pos-1]
			pos = pos - 1

		}
		sample[pos] = a

	}
	fmt.Println("\nAfter SelectionSort")
	for _, val := range sample {
		fmt.Print(val, " ")
	}
}

File: searching-sorting-in-python/insertion.py
def insertion(l):
    n=len(l)
    for i in range(1,n):
        cvalue=l[i]
        pos=i
        while pos>0 and l[pos-1]>cvalue:
            l[pos]=l[pos-1]
            pos=pos-1
        l[pos]=cvalue

l=[1, 0, -1, -1, 1, 0, 1, 0]
insertion(l)
print(l)